metsrw/fsentry.py

```diff
@@ -298,7 +300,43 @@ def add_rightsmd(self, md, mdtype, mode="mdwrap", **kwargs):
         return self._add_metadata_element(md, "rightsMD", mdtype, mode, **kwargs)
 
     def add_dmdsec(self, md, mdtype, mode="mdwrap", **kwargs):
-        return self._add_metadata_element(md, "dmdSec", mdtype, mode, **kwargs)
+        """Add dmdsec.
+
+        Extension of _add_metadata_element that adds a dmdSec and updates the
+        previous dmdSecs of the same mdtype_othermdtype, marking them as
+        "superseded" and using the same group_id for all of them.
+        """
+        dmdsec = self._add_metadata_element(md, "dmdSec", mdtype, mode, **kwargs)
+        dmdsec.status = kwargs.get("status") or "original"
+        mapping_key = mdtype + "_" + kwargs.get("othermdtype", "")
+        if mapping_key in self.dmdsecs_mapping:
+            group_id = getattr(self.dmdsecs_mapping[mapping_key][0], "group_id")
+            if not group_id:
+                group_id = str(uuid4())
+            dmdsec.group_id = group_id
+            for previous_dmdsec in self.dmdsecs_mapping[mapping_key]:
+                previous_dmdsec.group_id = group_id
+                if not previous_dmdsec.status:
+                    previous_dmdsec.status = "original"
+                if not previous_dmdsec.status.endswith("-superseded"):
+                    previous_dmdsec.status += "-superseded"
+        self.dmdsecs_mapping.setdefault(mapping_key, []).append(dmdsec)
+        return dmdsec
+
+    def delete_dmdsec(self, mdtype, othermdtype=""):
+        """Mark latest dmdsec of mdtype_othermdtype as deleted.
+
+        It doesn't delete the dmdsec from the METS. It only sets its status
+        attribute to "deleted".
+        """
+        mapping_key = mdtype + "_" + othermdtype
+        if mapping_key in self.dmdsecs_mapping:
+            self.dmdsecs_mapping[mapping_key][-1].status = "deleted"
+
+    def has_dmdsec(self, mdtype, othermdtype=""):
+        """Check if a dmdsec of mdtype_othermdtype exists for this entry."""
+        mapping_key = mdtype + "_" + othermdtype
+        return mapping_key in self.dmdsecs_mapping
```

Comment:
@replaceafill ... I added this function only to maintain the logging logic in https://github.com/artefactual/archivematica/blob/dev/xml-metadata-poc/src/MCPClient/lib/clientScripts/archivematicaCreateMETSReingest.py#L202.

Not sure if you think it has value for the future but for now, if we could tweak the logging a bit, we could remove it.